ports/stm32/uart.c

```diff
@@ -75,8 +76,13 @@
 #define CHAR_WIDTH_8BIT (0)
 #define CHAR_WIDTH_9BIT (1)
 
+typedef enum {
+    IRQ_FLAG_RX_IDLE= 1<<0,
```

Comment:
It might be a good idea to make the flags the same as those in the ST datasheet.  That might simplify the handling of IRQs by simply copying the UART->SR register into the flags variable, then checking if any should trigger.  See below for more comments on this.